### PR TITLE
obs-ffmpeg: Fix for FFmpeg 6 deprecating a flag

### PR DESCRIPTION
**Pull request by pkviet**
### Description
Also modifies libobs & deps/media-playback.
AV_CODEC_CAP_TRUNCATED was removed by avcodec 60 [1]. We ifdef the code depending on it to allow compilation.

[1] avcodec: remove FF_API_FLAG_TRUNCATED
https://github.com/FFmpeg/FFmpeg/commit/3ceffe783965767e62d59e8e68ecd265c98460ec

### Motivation and Context
Allow compilation of obs with FFmpeg 6.
Fixes https://github.com/obsproject/obs-studio/issues/8375

### How Has This Been Tested?
Tested with FFmpeg master as of Feb. 28 2023 (67fd1b79 libavcodec/hevc: remove duplicate semicolon in hevcdsp_init_neon).
Obs compiles fine with the new dlls.
Checked that Media Source works.
Checked that recording works (to mkv).
Checked that remuxing works (to mp4).
Checked that streaming to Twitch still works.

### Types of changes
- Bug fix (non-breaking change which fixes an issue)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by tytan652** [comment]
It didn't deprecated it, it dropped it https://github.com/FFmpeg/FFmpeg/commit/3ceffe783965767e62d59e8e68ecd265c98460ec.

File: deps/media-playback/media-playback/decode.c [hunk 1]
```diff
@@ -230,8 +230,10 @@ bool mp_decode_init(mp_media_t *m, enum AVMediaType type, bool hw)
 		d->in_frame = d->sw_frame;
 	}
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 8, 100)
```
Comment (derrod) [inline deps/media-playback/media-playback/decode.c:233]: I'm pretty sure all of those could be simplified to `LIBAVCODEC_VERSION_MAJOR < 60`

**Comment by pkviet** [comment]
> It didn't deprecated it, it dropped it [FFmpeg/FFmpeg@3ceffe7](https://github.com/FFmpeg/FFmpeg/commit/3ceffe783965767e62d59e8e68ecd265c98460ec).

oh thanks, i'll correct

File: deps/media-playback/media-playback/decode.c (hunk 1 above)
Comment (pkviet) [reply deps/media-playback/media-playback/decode.c:233]: that's true, i see that in the commit Tytan has linked indeed

**Review by RytoEX — APPROVED** [review]
Looks good to me.

**Pull request merged**